lms/static/scripts/frontend_apps/components/Dialog.js

```diff
@@ -4,53 +4,45 @@ import propTypes from 'prop-types';
 import classNames from 'classnames';
 
 import Button from './Button';
+import useElementShouldClose from '../common/use-element-should-close';
 import { zIndexScale } from '../utils/style';
+import { useUniqueId } from '../utils/hooks';
 
 /**
- * Accessibility notes:
+ * A model dialog wrapper with a title. The wrapper sets initial focus to itself
+ * unless an element inside of it is specified with the `initialFocus` ref.
+ * Optional action buttons may be passed in with the `buttons` prop but the
+ * cancel button is automatically generated when the on `onCancel` function is
+ * passed.
  *
- * Dialog accessibility is not trivial. We may want to use something like
- * https://github.com/reactjs/react-modal instead or as the basis of this
- * component.
+ * Canonical resources:
  *
- * Resources:
- *
- * 1. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role
- * 2. https://developer.paciellogroup.com/blog/2018/06/the-current-state-of-modal-dialog-accessibility/
- * 3. https://www.marcozehe.de/2015/02/05/advanced-aria-tip-2-accessible-modal-dialogs/
- *
- * [3] is the most useful resource IMO as it highlights the essentials from
- * the perspective of a blind engineer.
+ * https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html
+ * https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/alertdialog.html
  *
  * Things that are not implemented here yet:
  *
  * - A description which is reliably read out when the dialog is opened, in
  *   addition to the title.
- * - Saving and restoring keyboard focus when dialog is mounted and unmounted.
- * - Keeping tab focus within the dialog when shown.
- * - Hiding content underneath the dialog from screen readers.
- */
-
-/**
- * A modal dialog with a title and a row of action buttons at the bottom.
  */
 export default function Dialog({
   children,
   contentClass,
   initialFocus,
   onCancel,
+  role = 'dialog',
   title,
   buttons,
 }) {
-  const handleKey = event => {
-    event.stopPropagation();
+  const dialogId = useUniqueId('dialog');
```

Comment:
This is used to refer to the _title_ of the dialog rather than the dialog itself, so perhaps `dialogTitleId` and adjust the prefix accordingly.